pallets/pips/src/lib.rs

```diff
@@ -809,194 +864,264 @@ decl_module! {
             Self::deposit_event(RawEvent::Voted(current_did, proposer, id, aye_or_nay, deposit));
         }
 
-        /// An emergency stop measure to kill a proposal. Governance committee can kill
-        /// a proposal at any time.
-        #[weight = (550_000_000, DispatchClass::Operational, Pays::Yes)]
-        pub fn kill_proposal(origin, id: PipId) {
+        /// Approves the pending non-cooling committee PIP given by the `id`.
+        ///
+        /// # Errors
+        /// * `BadOrigin` unless a GC voting majority executes this function.
+        /// * `NoSuchProposal` if the PIP with `id` doesn't exist.
+        /// * `IncorrectProposalState` if the proposal isn't pending.
+        /// * `ProposalOnCoolOffPeriod` if the proposal is cooling off.
+        /// * `NotByCommittee` if the proposal isn't by a committee.
+        #[weight = (1_000_000_000, DispatchClass::Operational, Pays::Yes)]
+        pub fn approve_committee_proposal(origin, id: PipId) {
+            // 1. Only GC can do this.
             T::VotingMajorityOrigin::ensure_origin(origin)?;
-            ensure!(<Proposals<T>>::contains_key(id), Error::<T>::NoSuchProposal);
-            // Check that the proposal is pending
+
+            // 2. Proposal must be pending.
             Self::is_proposal_state(id, ProposalState::Pending)?;
-            Self::refund_proposal(id);
-            Self::update_proposal_state(id, ProposalState::Killed);
-            Self::prune_data(id, Self::prune_historical_pips());
+
+            // 3. Proposal must not be cooling-off and must be by committee.
+            let meta = Self::proposal_metadata(id).ok_or_else(|| Error::<T>::NoSuchProposal)?;
+            let curr_block_number = <system::Module<T>>::block_number();
+            ensure!(meta.cool_off_until <= curr_block_number, Error::<T>::ProposalOnCoolOffPeriod);
+            ensure!(matches!(meta.proposer, Proposer::Committee(_)), Error::<T>::NotByCommittee);
+
+            // 4. All is good, schedule PIP for execution.
+            let current_did = Context::current_identity::<Identity<T>>().unwrap_or_default();
```

Comment:
We should probably use a systematic DID here (i.e. the GCs DID)

Reply:
(Presumably applies wherever `T::VotingMajorityOrigin` is used.)

Reply:
Yep